src/DotNet.Status.Web/GitHubUserTokenHandler.cs

```diff
@@ -88,6 +102,35 @@ protected override async Task<AuthenticateResult> HandleAuthenticateAsync()
             return AuthenticateResult.Success(ticket);
         }
 
+        private string ParseBasicAuth(string authParameter)
+        {
+            string value = null;
+            byte[] rentedBytes = _bytePool.Rent(1000);
+            if (Convert.TryFromBase64String(authParameter, rentedBytes, out int size))
+            {
+                Span<byte> basicBytes = rentedBytes.AsSpan(0, size);
+
+                char[] rentedChars = _charPool.Rent(1000);
+                int cb = Encoding.UTF8.GetChars(basicBytes, rentedChars);
+                Span<char> basicChars = rentedChars.AsSpan(0, cb);
+
+                int separatorIndex = basicChars.IndexOf(':');
+                if (separatorIndex != -1)
+                {
+                    value = new string(basicChars.Slice(separatorIndex + 1));
+                }
+
+                // Clear out the auth stuff, no reason to leave it in memory if we don't have to
+                basicChars.Fill('\0');
+                basicBytes.Fill(0);
+
+                _charPool.Return(rentedChars);
+            }
+
+            _bytePool.Return(rentedBytes);
```

Comment:
should there be some `try {} finally {}`s here to make sure we Return the arrays we rented?

Reply:
I actually intentionally didn't do that.  If one of "TryParseBase64" or "Utf8.GetString" throws, which they shouldn't do, we are clearly in a bad place, and something is very wrong.  We probably shouldn't return potentially messed up arrays back to the pool.

It's always safe to _not_ return them. Since that will just cause it to allocate a new one.  This does mean that if some user is throwing crap auth tokens (things that aren't utf8 strings) we'll keep allocating new arrays... but that felt safer to me.  (also avoids another pair of finally blocks)